### PR TITLE
[SR-10960] URLSessionConfiguration.background(withIdentifier) method implemented

### PR DESCRIPTION
**Pull request by karthikkeyan**
Bug: https://bugs.swift.org/browse/SR-10960

### CODE REVIEW

**Comment by karthikkeyan** [comment]
cc @pushkarnk

**Comment by pushkarnk** [comment]
@karthikkeyan Thanks for taking this up. I understand this is for the sake of API parity. But whether the concept of a background task should to be supported on Linux/Windows is a big question that needs a discussion. I guess, background tasks may be relevant in the Swift on Android context though. 

Before we start implementing bits of this, it may be a good idea a start a discussion and decide on if, and how, we are going to support background tasks on non-Darwin platforms. This is my opinion. I'm eager to know what you, and others, think about this.

P.S: I recall [this discussion](https://forums.swift.org/t/nsurlsession-libcurl/1790/3) from the very early days :)

**Comment by karthikkeyan** [comment]
> @karthikkeyan Thanks for taking this up. I understand this is for the sake of API parity. But whether the concept of a background task should to be supported on Linux/Windows is a big question that needs a discussion. I guess, background tasks may be relevant in the Swift on Android context though.
> 
> Before we start implementing bits of this, it may be a good idea a start a discussion and decide on if, and how, we are going to support background tasks on non-Darwin platforms. This is my opinion. I'm eager to know what you, and others, think about this.
> 
> P.S: I recall [this discussion](https://forums.swift.org/t/nsurlsession-libcurl/1790/3) from the very early days :)

@pushkarnk  Yes, I also had the same doubt, but I thought it wouldn't hurt to give it a try, so I had created this PR. We don't have to review or work on this PR if _now_ is not an appropriate time for it. Thanks for sharing thread link, I have also created a thread as well [here](https://forums.swift.org/t/urlsessionconfiguration-background-withidentifier-implementation/26024). I will follow both of them.

File: Foundation/URLSession/URLSessionConfiguration.swift [hunk 1]
```diff
@@ -156,7 +156,17 @@ open class URLSessionConfiguration : NSObject, NSCopying {
         return ephemeralConfiguration
     }
 
-    open class func background(withIdentifier identifier: String) -> URLSessionConfiguration { NSUnimplemented() }
+    open class func background(withIdentifier identifier: String) -> URLSessionConfiguration {
```
Comment (millenomi) [inline Foundation/URLSession/URLSessionConfiguration.swift]: We do not have a daemon handling background sessions outside of Darwin. This should be marked `@availability(*, unavailable)` with a message.

**Comment by millenomi** [comment]
If an API does something that has strong guarantees, we strive to implement the same guarantees. `.background(…)` creates sessions that the developer expects to complete after the process is killed, and we cannot guarantee that happens without the infrastructure Darwin provides. We should not attempt to make it available this way; instead, a user who is porting code that requires background sessions can sit down and evaluate if they are using background sessions because they rely on these semantics (and restructure their code to match the fact they aren't available) vs. using them because they benefit from those semantics but could still port this code if this was a regular URLSession.

File: Foundation/URLSession/URLSessionConfiguration.swift (hunk 1 above)
Comment (karthikkeyan) [reply Foundation/URLSession/URLSessionConfiguration.swift]: @millenomi  I have reverted the implementation and added `@available(*, unavailable, message: "Not available on non-Darwin platforms")`

**Comment by millenomi** [comment]
@swift-ci please test Linux

**Comment by millenomi** [comment]
@swift-ci please test Linux

**Pull request merged**